code/game/machinery/vending.dm

```diff
@@ -273,14 +277,14 @@
 					user << "<span class='notice'>You loaded [transfered] items in \the [name].</span>"
 				else
 					user << "<span class='notice'>The [name] is fully stocked.</span>"
-			return;
+			return
 		else
 			user << "<span class='notice'>You should probably unscrew the service panel first.</span>"
 	else
-		..()
+		return ..()
 
 
-/obj/machinery/vending/default_deconstruction_crowbar(obj/item/O)
+/obj/machinery/vending/deconstruction()

```

Comment:
deconstruction() is called by default_deconstruction_crowbar() on success, so no need to override it.